homeassistant/components/ios/storage.py

```diff
@@ -0,0 +1,81 @@
+"""Storage for iOS CarPlay configuration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.storage import Store
+from homeassistant.util.hass_dict import HassKey
+
+DATA_CARPLAY_STORAGE: HassKey[CarPlayStore] = HassKey("ios_carplay_storage")
+STORAGE_VERSION = 1
+STORAGE_KEY = "ios.carplay_config"
+
+
+class CarPlayStore:
+    """Store for CarPlay configuration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the CarPlay store."""
+        self.hass = hass
+        self._store = Store[dict[str, Any]](hass, STORAGE_VERSION, STORAGE_KEY)
+        self.data: dict[str, Any] = {}
+        self.subscriptions: dict[str | None, list[Callable[[], None]]] = {}
+
+    async def async_load(self) -> None:
+        """Load the data from storage."""
+        stored_data = await self._store.async_load()
+        if stored_data is not None:
+            self.data = stored_data
+        else:
+            # Default carplay configuration
+            self.data = {
```

Comment:
I'd use a dataclass instead so we can have typed easily accessible attributes.

Alternatively a typed dict.

https://docs.python.org/3/library/dataclasses.html